src/lib.rs

```diff
@@ -24,21 +24,27 @@
 #![allow(suspicious_double_ref_op)]
 
 use num::ToPrimitive;
+use serde::Serialize;
 use std::cmp::Ordering;
 //use float_eq::AssertFloatEq;
 // use std::collections::HashMap;
 
 /// Our main data struct
-#[derive(Clone, Default, Debug, PartialEq)]
+#[derive(Clone, Default, Debug, PartialEq, Serialize)]
 pub struct SimpleAccumulator {
+    #[serde(skip_serializing)]
     /// Vec to store the data
     pub vec: Vec<f64>,
+    #[serde(skip_serializing)]
     /// Vec to privately store mean and three moment differences
     stats: Vec<f64>,
+    #[serde(skip_serializing)]
```

Comment:
All `pub` should be serialized. Why we have different `r_mean` and `mean`? Both are always equal. And we should always have only running mean. It an accumulator.

Reply:
We have both of them for the `--fixed-capacity` case. They will be equal when capacity is not fixed.

In the `--fixed-capacity` case, we need `mean` of the local data/data in the accumulator to calculate the `bimodality` measure on the data in the accumulator at that moment. I am unsure if `r_mean` or running mean should be used for that.

Reply:
I don't completely follow. They (`mean` and `mean_r`) must be equal whether or not capacity is fixed. Capacity is not fixed because we don't want to/can't store the whole history.  The running mean algorithm is guaranteed to be equal to the mean calculated with the standard formula. 

If we are computing another "mean" to estimate the bimodality then it should be private and should be renamed to something else e.g. `mean_of_stored_history`  and should not be exposed as the "mean" of data in public API. 

Ref: https://github.com/joshday/OnlineStats.jl